File: 111-minimumDepthOfBinaryTree/minDepth.js
"use strict";
function minDepth(root) {
    if (root === null) {
        return 0;
    }
    const Q = [root];
    let depth = 1;
    while (Q.length !== 0) {
        const layerSize = Q.length;
        for (let i = 0; i < layerSize; i++) {
            const node = Q.shift();
            if (node.left === null && node.right === null) {
                return depth;
            }
            if (node.left !== null) {
                Q.push(node.left);
            }
            if (node.right !== null) {
                Q.push(node.right);
            }
        }
        depth++;
    }
    return -1; // unreachable
}
;

File: 111-minimumDepthOfBinaryTree/minDepth.go
package minDepth

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func minDepth(root *TreeNode) int {
	if root == nil {
		return 0
	}
	Q := []*TreeNode{root}
	depth := 1
	for {
		layerSize := len(Q)
		for i := 0; i < layerSize; i++ {
			node := Q[0]
			Q = Q[1:]
			if node.Left == nil && node.Right == nil {
				return depth
			}
			if node.Left != nil {
				Q = append(Q, node.Left)
			}
			if node.Right != nil {
				Q = append(Q, node.Right)
			}
		}
		depth++
	}
}
